### PR TITLE
Feature/user ui

### PR DESCRIPTION
**Pull request by ashay123**
#99, 

# Introduction
Added a window at the very beginning of the app which asks for the Name and IBAN of the user. This user is put in the User class, which for now isn't integrated in the whole program, but this will be done in a upcoming PR.

# Purpose
Now the program keeps the information in memory, so it can be used throughout all the activities, instead of a hardcoded way.

# Future Applicability
Each user of the app now gets a unique chain based on his/her own information.

# PR Reflection

### CODE REVIEW

File: app/src/main/java/nl/tudelft/b_b_w/view/ContactsActivity.java [hunk 1]
```diff
@@ -44,9 +31,8 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_contacts);
         setTitle("Contacts");
-        Bundle extras = getIntent().getExtras();
-        ownerName = extras.getString("ownerName");
-        // get contacts
+        SharedPreferences settings = getSharedPreferences(MainActivity.PREFS_NAME, 0);
+        final String ownerName = settings.getString("userName", "");
         BlockController blcController = new BlockController(this);
```
Comment (humanjas) [inline app/src/main/java/nl/tudelft/b_b_w/view/ContactsActivity.java]: Don't use abbreviations.

File: app/src/main/java/nl/tudelft/b_b_w/view/MainActivity.java [hunk 2]
```diff
@@ -75,21 +91,26 @@ public void onClick(DialogInterface dialog, int which) {
...
-                    blockController.getLatestSeqNumber(ownerName) + 1,
-                    cvc.hashKey(),
-                    "N/A",
-                    "N/A",
-                    "pubkey",
-                    "Iban",
-                    0
-            );
-            blockController.addBlock(block);
+                ConversionController cvc = new ConversionController(
```
Comment (humanjas) [inline app/src/main/java/nl/tudelft/b_b_w/view/MainActivity.java]: Use the function in the blockcontroller to create blocks.

File: app/src/main/java/nl/tudelft/b_b_w/view/ContactAdapter.java [hunk 3]
```diff
@@ -73,10 +73,14 @@ public int getCount() {
      * @return Image number
```
Comment (deepweeb) [inline app/src/main/java/nl/tudelft/b_b_w/view/ContactAdapter.java]: Forgot @param

File: app/src/main/java/nl/tudelft/b_b_w/view/ContactAdapter.java [hunk 4]
```diff
@@ -73,10 +73,14 @@ public int getCount() {
...
      */
     private int getImageNo(int trust) {
-        if (trust > 100) trust = 100; //maximum value
-        if (trust < 0) trust = 0; //minimum value
+        return images[calculateImageIndex(trust)];
+    }
+
+    private int calculateImageIndex(int trust) {
```
Comment (deepweeb) [inline app/src/main/java/nl/tudelft/b_b_w/view/ContactAdapter.java]: Forgot Comments

File: app/src/main/java/nl/tudelft/b_b_w/view/ContactAdapter.java (hunk 3 above)
Comment (ashay123) [inline app/src/main/java/nl/tudelft/b_b_w/view/ContactAdapter.java]: Did not

Comment (deepweeb) [inline app/src/main/java/nl/tudelft/b_b_w/view/ContactAdapter.java]: Sorry forgot to expand the code

**Pull request merged**